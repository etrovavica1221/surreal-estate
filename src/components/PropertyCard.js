/* eslint-disable react/jsx-one-expression-per-line */
/* eslint-disable global-require */
/* eslint-disable import/no-duplicates */
/* eslint-disable react/require-default-props */
import React from 'react';
import PropTypes from 'prop-types';
import '../styles/PropertyCard.css';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faBed } from '@fortawesome/free-solid-svg-icons';
import { faBath } from '@fortawesome/free-solid-svg-icons';
import { faEnvelope } from '@fortawesome/free-solid-svg-icons';

function numberWithCommas(x) {
  return x.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ',');
}

const PropertyCard = ({ title, city, type, bedrooms, bathrooms, price, email }) => (
  <div className="Property">
    <div id="house-padding">
      <img id="house-icon" src={require('../styles/house-icon.svg')} alt="house icon" width="60" />
    </div>
    <div className="items" id="title">
      {title}
    </div>
    <div className="items" id="city">
      {type} - {city}
    </div>
    <div className="items">
      <FontAwesomeIcon icon={faBed} />
      {bedrooms}
    </div>
    <div className="items">
      <FontAwesomeIcon icon={faBath} />
      {bathrooms}
    </div>
    <div className="items">
      Â£{numberWithCommas(price)}
    </div>
    <div className="items" id="email">
      <a href={`mailto: ${email}`}>
        <FontAwesomeIcon icon={faEnvelope} />
        email
      </a>
    </div>
  </div>
);

PropertyCard.propTypes = {
  title: PropTypes.string,
  city: PropTypes.string,
  type: PropTypes.string,
  bedrooms: PropTypes.string,
  bathrooms: PropTypes.string,
  price: PropTypes.string,
  email: PropTypes.string,
};

export default PropertyCard;
