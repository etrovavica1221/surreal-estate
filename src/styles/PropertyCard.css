.Property {
  display: flex;
  flex-direction: column;
  width: 16em;
  padding: 15px;
  margin: auto;
  border-style: groove;
  border-color: rgb(145, 64, 202);
  border-width: 3px;
  background-color: lavender;
  justify-content: space-around;
  align-items: stretch;
}

#title {
  font-weight: bold;
}

#city {
  font-style: italic;
}

#email,
#save-btn,
#delete-btn {
  display: flex;
  justify-content: center;
  background-color: rgb(93, 78, 187);
  margin: 10px 0;
  padding: 5px;
  color: white;
  font-size: 18px;
  text-shadow: -1px 0 black, 0 1px black, 1px 0 black, 0 -1px black;
  border-style: groove;
  border-color: rgb(145, 64, 202);
  border-width: 3px;
  border-radius: 6px;
  text-decoration: none;
}

#save-btn,
#delete-btn {
  width: 100%;
  cursor: pointer;
}

.Property a {
  text-decoration: none;
}

.items {
  min-height: 20px;
}

#house-icon {
  margin: 0 auto 0 auto;
  width: 4em;
  justify-self: center;
  background-color: rgb(235, 227, 222);
  padding: 5em;
}

#house-padding {
  justify-self: center;
  display: flex;
  justify-self: center;
  margin: 0 auto 30px auto;
  background-color: rgb(235, 227, 222);
}

@media only screen and (max-width: 350px) {
  #house-icon {
      margin: 0 auto 0 auto;
      width: 4em;
      justify-self: center;
      background-color: rgb(235, 227, 222);
      padding: 4em;
  }

  .Property {
    width: 13em;
  }
}