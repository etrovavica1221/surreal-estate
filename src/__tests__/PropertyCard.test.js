import React from 'react';
import { render } from '@testing-library/react';
import PropertyCard from '../components/PropertyCard';

it('renders properties', () => {
  const { getByText, asFragment } = render(<PropertyCard />);
  expect(asFragment()).toMatchSnapshot();
  expect(getByText(/email/).textContent).toBe('email');
});
