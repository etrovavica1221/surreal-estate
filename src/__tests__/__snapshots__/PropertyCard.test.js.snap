// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders properties 1`] = `
<DocumentFragment>
  <div
    class="Property"
  >
    <div
      id="house-padding"
    >
      <img
        alt="house icon"
        id="house-icon"
        src="[object Object]"
        width="60"
      />
    </div>
    <div
      class="items"
      id="title"
    />
    <div
      class="items"
      id="city"
    >
       - 
    </div>
    <div
      class="items"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-bed fa-w-20 "
        data-icon="bed"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 640 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M176 256c44.11 0 80-35.89 80-80s-35.89-80-80-80-80 35.89-80 80 35.89 80 80 80zm352-128H304c-8.84 0-16 7.16-16 16v144H64V80c0-8.84-7.16-16-16-16H16C7.16 64 0 71.16 0 80v352c0 8.84 7.16 16 16 16h32c8.84 0 16-7.16 16-16v-48h512v48c0 8.84 7.16 16 16 16h32c8.84 0 16-7.16 16-16V240c0-61.86-50.14-112-112-112z"
          fill="currentColor"
        />
      </svg>
    </div>
    <div
      class="items"
    >
      c
    </div>
    <div
      class="items"
    >
      Â£
    </div>
    <div
      class="items"
      id="email"
    >
      <a
        href="mailto: undefined"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-envelope fa-w-16 "
          data-icon="envelope"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M502.3 190.8c3.9-3.1 9.7-.2 9.7 4.7V400c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V195.6c0-5 5.7-7.8 9.7-4.7 22.4 17.4 52.1 39.5 154.1 113.6 21.1 15.4 56.7 47.8 92.2 47.6 35.7.3 72-32.8 92.3-47.6 102-74.1 131.6-96.3 154-113.7zM256 320c23.2.4 56.6-29.2 73.4-41.4 132.7-96.3 142.8-104.7 173.4-128.7 5.8-4.5 9.2-11.5 9.2-18.9v-19c0-26.5-21.5-48-48-48H48C21.5 64 0 85.5 0 112v19c0 7.4 3.4 14.3 9.2 18.9 30.6 23.9 40.7 32.4 173.4 128.7 16.8 12.2 50.2 41.8 73.4 41.4z"
            fill="currentColor"
          />
        </svg>
        email
      </a>
    </div>
  </div>
</DocumentFragment>
`;
